hw/drivers/sensors/lis2dw12/src/lis2dw12.c

```diff
@@ -2052,18 +2332,22 @@ lis2dw12_sensor_unset_notification(struct sensor *sensor, sensor_event_type_t ty
     struct lis2dw12 * lis2dw12;
     struct sensor_itf *itf;
     int rc;
+    uint8_t int_cfg = 0;
     
-    if ((type & ~(SENSOR_EVENT_TYPE_DOUBLE_TAP |
-                  SENSOR_EVENT_TYPE_SINGLE_TAP)) != 0) {
-        return SYS_EINVAL;
+    if(type == SENSOR_EVENT_TYPE_DOUBLE_TAP) {
+        int_cfg |= LIS2DW12_INT1_CFG_DOUBLE_TAP;
     }
-    
-    /*XXX for now we do not support registering for both events */
-    if (type == (SENSOR_EVENT_TYPE_DOUBLE_TAP |
-                 SENSOR_EVENT_TYPE_SINGLE_TAP)) {
+    else if(type == SENSOR_EVENT_TYPE_SINGLE_TAP) {
```

Comment:
As per our coding style, <else if> should be on the same line as < } > and < { >.